packages/web-components/src/components/va-number-input/va-number-input.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  Component,
+  Element,
+  Prop,
+  Host,
+  h,
+  Event,
+  EventEmitter
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-number-input',
+  styleUrl: 'va-number-input.css',
+  shadow: true,
+})
+export class VaNumberInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the text input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean;
+
+  /**
+   * The inputmode attribute.
+   */
+  @Prop() inputmode?: 'decimal' | 'numeric';
+
+  /**
+   * Emit component-library-analytics events on the blur event.
+   */
+  @Prop() enableAnalytics?: boolean;
+
+  /**
+   * The name to pass to the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The aria-describedby attribute for the `<input>` in the shadow DOM.
+   */
+  @Prop() ariaDescribedby?: string = '';
+
+  /**
+   * Minimum number value
+   * The min attribute specifies the minimum value for an `<input>` element.
+   */
+  @Prop() min: number | string;
+
+  /**
+   * Maximum number value
+   * The max attribute specifies the maximum value for an `<input>` element.
+   */
+  @Prop() max: number | string;
+
+  /**
+   * The value for the input.
+   */
+  @Prop({ mutable: true }) value?: string;
+  // TODO: Make the value prop reflective. Currently, it isn't because it screws
+  // up the input behavior. For now, the only "bug" is that the changed value
+  // isn't reflected in the DOM on the web component. That seems to be how the
+  // <input> is supposed to work, however:
+  // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-value
+  //
+  // $('va-number-input').value will be correct
+  // $('va-number-input').getAttribute('value') will be incorrect
+
+  /**
+   * The event emitted when the input is blurred.
+   */
+  @Event() vaBlur: EventEmitter;
+
+  /**
+   * The event emitted when the input value changes
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * input is blurred and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.value = target.value;
+    this.vaChange.emit({ value: this.value });
+  };
+
+  private handleBlur = () => {
+    this.vaBlur.emit();
```

Comment:
Would it be possible to use a native event instead of a custom one? The [`blur` event is composed but it doesn't bubble](https://developer.mozilla.org/en-US/docs/Web/API/Element/blur_event), but I'm not sure if the no bubbling would matter here. For example:

```jsx
<va-number-input onBlur={(e) => console.log('blur event', e)} />
```

The `onBlur` handler should still fire, but when the web component is blurred instead of the inner `<input>`. The effect might still be the same, but we'd need to test to know for sure.